cli/generate_permissions.py

```diff
@@ -28,48 +28,77 @@
     from enum import Enum
     from typing import Dict, List
 
+    from ..shared.exceptions import PermissionException
     from .get_permission_parts import get_permission_parts
 
 
-    class OrganisationManagementLevel(str, Enum):
+    class CompareRightLevel(str):
+        def get_number(self, value) -> int:  # type: ignore
+            return 0
+
+        def __lt__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number < other_number
+
+        def __le__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number <= other_number
+
+        def __gt__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number > other_number
+
+        def __ge__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number >= other_number
+
+
+    class OrganisationManagementLevel(CompareRightLevel, Enum):
         SUPERADMIN = "superadmin"
         CAN_MANAGE_USERS = "can_manage_users"
         CAN_MANAGE_ORGANISATION = "can_manage_organisation"
-
-        def __init__(self, oml: str):
-            super().__init__()
-            self.numbers = {
+        NO_RIGHT = "no_right"
+
+        def get_number(self, value: "OrganisationManagementLevel") -> int:
+            if not isinstance(value, self.__class__):
+                raise PermissionException(
+                    f"The comparison expect an {self.__class__}-type and no string!"
+                )
+            numbers = {
                 "superadmin": 3,
                 "can_manage_organisation": 2,
                 "can_manage_users": 1,
+                "no_right": 0,
             }
-            self.number: int = self.numbers.get(oml, 0)
-
-        def is_ok(self, user_oml: str) -> bool:
-            return self.numbers.get(user_oml, 0) >= self.number
-
+            return numbers.get(value, 0)
```

Comment:
You can implement this method in the base class and just set the `numbers` as a class variable.

Reply:
Done, although because of base class Enum this is not as easy